### PR TITLE
Immediately send progress on disable submit

### PR DESCRIPTION
**Pull request by e-kolpakov**
Concern: modifications to `light_children.py` contains in-function import, which is added to break circular import. There reason why `MentoringBlock` is needed there is to fix a subtle bug in how light_children parses XML and sets attributes on blocks.

In short it was skipping XBlock Fields' deserialization and just assigned string values to block attributes - hence display_submit worked incorrectly, as string "false" is true-ish in boolean context. The workaround is to check if we're setting data for `MentoringBlock` (as the only XBlock with real XBlock Fields) and force parsing the value with appropriate Field.

Since we have mentoring-v2 (aka "real chilrden") almost ready, `light_children` modifications should be viewed as a temporary workaround for a problem, as mentoring-v2 would not be subject to this issue.

### CODE REVIEW

File: mentoring/light_children.py [hunk 1]
```diff
@@ -101,12 +101,17 @@ def _is_default(value):
...
     def init_block_from_node(cls, block, node, attr):
+        from mentoring import MentoringBlock
         block.light_children = []
         for child_id, xml_child in enumerate(node):
             cls.add_node_as_child(block, xml_child, child_id)
 
         for name, value in attr:
-            setattr(block, name, value)
+            if isinstance(block, MentoringBlock):
+                parsed_value = getattr(MentoringBlock, name).from_json(value)
```
Comment (antoviaque) [inline mentoring/light_children.py]: This would change the behavior - you're getting the attribute from the field instead of getting it from the function parameter. I would be very wary of making such a change without fully understanding the effects.

Also, if the goal is to ensure you do have a boolean, you don't need to change anything here - there is an XBlock field property we have contributed specifically for this type of issue, `enforce_type` - cf https://github.com/edx/XBlock/pull/200 . See https://github.com/edx-solutions/xblock-mentoring/blob/f4f37ca9a39dfdf6c58db0e3b1563738372bf13e/mentoring/mentoring.py#L99 for an example.

File: mentoring/light_children.py [hunk 2]
```diff
@@ -101,12 +101,17 @@ def _is_default(value):
...
     @classmethod
     def init_block_from_node(cls, block, node, attr):
+        from mentoring import MentoringBlock
         block.light_children = []
         for child_id, xml_child in enumerate(node):
             cls.add_node_as_child(block, xml_child, child_id)
 
         for name, value in attr:
-            setattr(block, name, value)
+            if isinstance(block, MentoringBlock):
```
Comment (antoviaque) [inline mentoring/light_children.py]: Minor given the other comment below, but this file shouldn't contain anything specific to Mentoring at this level -- if you want to check whether this is a real XBlock, you should check for the XBlock class, not MentoringBlock.

File: mentoring/light_children.py (hunk 1 above)
Comment (e-kolpakov) [inline mentoring/light_children.py]: @antoviaque actually it fixed a couple of other bugs: e.g. submit was still shown, but disabled when display_submit="false". I'll give `enforce_type` a shot, but if I understand right `setattr` just replaced whatever `display_submit` was with a plain `str`.

**Comment by antoviaque** [comment]
The change looks good, thanks! I feel much safer now : )

Testing locally also seemed to work well - LGTM :+1:

File: mentoring/mentoring.py [hunk 3]
```diff
@@ -97,7 +97,8 @@ def is_default_xml_content(value):
                       default='mentoring-default', scope=Scope.content)
     enforce_dependency = Boolean(help="Should the next step be the current block to complete?",
                                  default=False, scope=Scope.content, enforce_type=True)
-    display_submit = Boolean(help="Allow to submit current block?", default=True, scope=Scope.content)
+    display_submit = Boolean(help="Allow to submit current block?", default=True,
```
Comment (Kelketek) [inline mentoring/mentoring.py]: Can we rephrase this to 'Allow submission of the current block?'

**Comment by e-kolpakov** [comment]
@Kelketek done as well

**Comment by antoviaque** [comment]
Merging to get it to QA, but @Kelketek this is missing an explicit :+1: which would have allowed @e-kolpakov to merge it himself. Can you confirm this is good for you and tested, and post the :+1: ? Thanks!
